syntax = "proto3";
package dracon.v1;
option go_package = "v1";

import "issue.proto";
import "config.proto";
import "google/protobuf/timestamp.proto";

/*
    Scan related information, unique and immutable per scan run
    This message is copied from LaunchToolRequest to LaunchToolResponse
    by each producer wrapper
*/
message ScanInfo {
    // scan unique identifier
    string scan_uuid = 1;
    // timestamp of when the scan was triggered (passed to LaunchToolResponse)
    google.protobuf.Timestamp scan_start_time = 2;
}

/* LaunchToolRequest consists of a request configuration,
 * built by the engine and processed by a producer,
 * through which options are passed */
message LaunchToolRequest {
    ScanInfo scan_info = 1;
    dracon.v1.Config config = 2;
}

/* LaunchToolReponse consists of a response built by a producer,
 * to be interpreted by a consumer */
message LaunchToolResponse {
    ScanInfo scan_info = 1;
    // The name of the tool that ran the scan
    string tool_name = 2;
    // Issues discovered during the scan
    repeated dracon.v1.Issue issues = 3;
}

/*
    An EnrichedLaunchToolResponse consists of deduplicated vulnerability
    information, with added metadata for consumers
*/
message EnrichedLaunchToolResponse {
    // The results of the original scan prior to enrichment
    LaunchToolResponse original_results = 1;
    // Enriched, deduplicated issues
    repeated EnrichedIssue issues = 2;
}
